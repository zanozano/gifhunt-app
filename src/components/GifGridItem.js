import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ title, url }) => {
  // console.log(id, title, url);

  return (
    <div className='card animate__animated animate__fadeIn'>
      <img src={url} alt={title} />
      <h4>{title}</h4>
    </div>
  );
};

//* propTypes are required
GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
