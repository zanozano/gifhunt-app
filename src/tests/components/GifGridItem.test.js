import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifGridItem from '../../components/GifGridItem';

//*Testing
describe('Components testing >>', () => {
  const title = 'Un titulo';
  const url = 'https://localhost/';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Testing component <GifGridItem />', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should have a paragraph with a title', () => {
    const p = wrapper.find('h4').text().trim();
    console.log(p);

    expect(p).toBe(title);
  });
  test('Should have a src img and alt', () => {
    const img = wrapper.find('img').html();
    const imgProps = wrapper.find('img').prop('src');
    const imgAlt = wrapper.find('img').prop('alt');

    //* Option 1 wrapper.find('img').props();
    //* Option 2 wrapper.find('img').prop('src');
    //* Option 3 wrapper.find('img').props().src;

    console.log(img);
    console.log(imgProps);

    expect(imgProps).toBe(url);
    expect(imgAlt).toBe(title);
  });
  test('Should have the classes fadeIn', () => {
    const div = wrapper.find('div').prop('className');
    expect(div.includes('animate__fadeIn')).toBe(true);
  });
});
