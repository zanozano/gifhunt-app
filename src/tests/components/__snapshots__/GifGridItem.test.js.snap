// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components testing >> Testing component <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/"
  />
  <h4>
    Un titulo
  </h4>
</div>
`;
